File: VB/WizardCustomizationExample1/DataSourceWizardCustomization1.vb
﻿Imports Microsoft.VisualBasic
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardWin.ServiceModel
Imports DevExpress.DataAccess.UI.Wizard
Imports DevExpress.DataAccess.Wizard.Presenters
Imports DevExpress.DataAccess.Wizard.Views
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace WizardCustomizationExample1
    Public Class DataSourceWizardCustomization
        Implements IDashboardDataSourceWizardCustomization

        Public Sub CustomizeDataSourceWizard(ByVal customization As IWizardCustomization(Of DashboardDataSourceModel)) Implements IDashboardDataSourceWizardCustomization.CustomizeDataSourceWizard
            customization.StartPage = GetType(ChooseConnectionPage(Of DashboardDataSourceModel))
            customization.Model.DataSourceType = DashboardDataSourceType.Xpo
            customization.RegisterPageView(Of IChooseConnectionPageView, CustomChooseConnectionPageView)()
        End Sub
    End Class
End Namespace

File: CS/WizardCustomizationExample1/DataSourceWizardCustomization1.cs
using DevExpress.DashboardCommon;
using DevExpress.DashboardWin.ServiceModel;
using DevExpress.DataAccess.UI.Wizard;
using DevExpress.DataAccess.Wizard.Presenters;
using DevExpress.DataAccess.Wizard.Views;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WizardCustomizationExample1 {
    public class DataSourceWizardCustomization : IDashboardDataSourceWizardCustomization {

        public void CustomizeDataSourceWizard(IWizardCustomization<DashboardDataSourceModel> customization) {
            customization.StartPage = typeof(ChooseConnectionPage<DashboardDataSourceModel>);
            customization.Model.DataSourceType = DashboardDataSourceType.Xpo;
            customization.RegisterPageView<IChooseConnectionPageView, CustomChooseConnectionPageView>();
        }
    }
}
